src/main/java/org/jenkinsci/plugins/docker/commons/credentials/DockerServerCredentials.java

```diff
@@ -66,8 +74,8 @@ public DockerServerCredentials(CredentialsScope scope, String id, String descrip
      * @return null if there's no authentication
      */
     @CheckForNull
-    public String getClientKey() {
-        return clientKey == null ? null : clientKey.getPlainText();
+    public Secret getClientKey() {
```

Comment:
It breaks binary compatibility.
Usage example: https://github.com/jenkinsci/docker-swarm-plugin/blob/aab115b35c59758ad6fc38ca01fee1f11fbfa2a0/src/main/java/org/jenkinsci/plugins/docker/swarm/DockerServerCredentialsSSLConfig.java#L29

Reply:
What is this CertificateUtils? Not part of docker-commons.

Reply:
Also, the alternative is to leave the getter as is and break data bound constructor.

Reply:
Do not break databinding please. You could leave the existing getter, deprecated, and introduce a new property with a different name.

Reply:
By renaming the property, we get a different JCasC syntax potentially. Or am I confusing something?

Reply:
I'm renaming the property to `clientKeySecret` while leaving the constructor and getter for ABI compatibility. Those are now deprecated.